onnxruntime/core/providers/webgpu/shader_helper.cc

```diff
@@ -408,12 +413,22 @@ Status ShaderHelper::GenerateSourceCode(std::string& code, std::vector<int>& sha
   //
   // Input/output variables
   //
-  size_t variable_count = 0;
-  for (const auto& input : input_vars_) {
-    ss << "@group(0) @binding(" << variable_count++ << ") var<storage, read> " << input->name_ << ": array<" << input->StorageType() << ">;\n";
+  for (size_t i = 0; i < input_vars_.size(); ++i) {
+    const auto& input = input_vars_[i];
+    ss << "@group(0) @binding(" << i << ") var<storage, read> " << input->name_ << ": array<" << input->StorageType() << ">;\n";
```

Comment:
Do we need to use `input_vars_.size() + i` instead of `i` to make it unique?

Reply:
thanks for catching. updated with fix.